asv_bench/benchmarks/groupby.py

```diff
@@ -987,4 +988,26 @@ def time_sample_weights(self):
         self.df.groupby(self.groups).sample(n=1, weights=self.weights)
 
 
+class Resample:
+    # GH 28635
+    def setup(self):
+        num_timedeltas = 100_000
+        num_groups = 5
+
+        index = MultiIndex.from_product(
+            [
+                np.arange(num_groups),
+                to_timedelta(np.arange(num_timedeltas), unit="s"),
+            ]
+        )
+
+        self.df = DataFrame(np.random.randint(0, 1000, size=(len(index))), index=index)
+
+    def time_resample(self):
+        self.df.reset_index(1).groupby(level=0).resample("10s", on="level_1").mean()
```

Comment:
Could you prepare df as needed in setup so that the time_* function only calls the function we want to measure?

Reply:
Ah yes, that makes sense, I did not consider that. I moved the reset_index() into the setup method. I also added names for the index levels, because the whole level=0, level=1 felt extra confusing after that change.

Reply:
You could also move the groupby call I think

Reply:
Oh I see, I thought since this is in file groupby.py we are measuring the perfomance of the combination of the groupby and resample operation? The class just before my addition "Sample" also has the groupby inside the time_ methods